### PR TITLE
KFLUXINFRA-1064: testing phase added to merged main branch SeaLights workflow and refactoring for DRY

### PR DESCRIPTION
**Pull request by meyrevived**
SeaLights asked that we'll add a testing phase to the github workflow that runs a SeaLights code scan on the main branch after it's been merged with new code.

This basically made sealights-mpc-test-ci.yaml extremely similar to this workflow and so I removed all the repeating code to a separate bash script that needs to be given flags depending on where it's run (pull request or merged main) and now the workflows just run this script with its flags.

### CODE REVIEW

File: mpc_testing_with_sealights.sh [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+      ;;
+  esac
+done
+
+# Validate mode and required parameters
+if [[ -z "$MODE" ]]; then
+  echo "Error: --mode is required" >&2
+  exit 1
+fi
+
```
Comment (ifireball) [inline mpc_testing_with_sealights.sh:45]: I would separate downloading dependencies and running processes, so that GitHub Actions` cache can be used

Comment (meyrevived) [reply mpc_testing_with_sealights.sh:45]: SeaLights are working on fetching their agent via go releases currently and I'm checking in on their progress on a daily basis. Using github actions cache would still be identical in both workflows and wouldn't be DRY if I separated them. I'm not sure we should invest in a different method for the moment, if a different solution is about to come out any day now. SeaLights are based in Israel and have been working through this holiday season so progress is being made.

**Comment by ifireball** [comment]
@meyrevived do we still need this PS given that #372 was merged?

**Comment by meyrevived** [comment]
Made differently in https://github.com/konflux-ci/multi-platform-controller/pull/372

**Pull request closed without merging**